pkg/utils/utils.go

```diff
@@ -176,9 +176,14 @@ func doMerge(src, dest reflect.Value) error {
 
 	case reflect.Map:
 		for _, key := range src.MapKeys() {
-			if err := doMerge(src.MapIndex(key), dest.MapIndex(key)); err != nil {
-				return err
+			srcElem := src.MapIndex(key)
+			if !srcElem.IsValid() || isEmptyValue(srcElem) {
+				continue
+			}
+			if dest.IsNil() {
```

Comment:
@Ace-Tang Add a test case to cover this statement?
like
```go
{
	src:      &simple{Sd: map[string]string{"go": "gogo"}},
	dest:     &simple{},
	expected: &simple{Sd: map[string]string{"go": "gogo"}},
	ok:       true,
},
```

Reply:
Already add the test. @zhuangqh

Reply:
cri-test alway fails to pull image.